tensorflow/lite/micro/kernels/reduce_test.cc

```diff
@@ -74,20 +74,96 @@ void TestMeanFloatInput4D(const int* input_dims_data, const float* input_data,
+
+  constexpr int tensors_size = num_of_inputs + num_of_outputs;
+  TfLiteTensor tensors[tensors_size] = {
+      CreateFloatTensor(input_data, input_dims),
+      CreateInt32Tensor(axis_data, axis_dims),
+      CreateFloatTensor(output_data, output_dims),
+  };
+
+  TF_LITE_MICRO_EXPECT_EQ(
+      kTfLiteOk, ValidateReduceGoldens(
+                     tensors, tensors_size, expected_output_data, output_data,
+                     output_dims_count, registration, params, tolerance));
+}
+
+void TestReduceOpFloat(const int* input_dims_data, const float* input_data,
+                       const int* axis_dims_data, const int32_t* axis_data,
+                       const int* output_dims_data,
+                       const float* expected_output_data,
+                       TfLiteRegistration registration,
+                       TfLiteReducerParams* params, float tolerance = 1e-5) {
+  TfLiteIntArray* input_dims = IntArrayFromInts(input_dims_data);
+  TfLiteIntArray* axis_dims = IntArrayFromInts(axis_dims_data);
+  TfLiteIntArray* output_dims = IntArrayFromInts(output_dims_data);
+  const int output_dims_count = ElementCount(*output_dims);
+
   constexpr int num_of_inputs = 2;   // input and axis
   constexpr int num_of_outputs = 1;  // output
 
+  float output_data[output_dims_count];
+
   constexpr int tensors_size = num_of_inputs + num_of_outputs;
   TfLiteTensor tensors[tensors_size] = {
       CreateFloatTensor(input_data, input_dims),
       CreateInt32Tensor(axis_data, axis_dims),
       CreateFloatTensor(output_data, output_dims),
   };
 
+  TF_LITE_MICRO_EXPECT_EQ(
+      kTfLiteOk, ValidateReduceGoldens(
+                     tensors, tensors_size, expected_output_data, output_data,
+                     output_dims_count, registration, params, tolerance));
+}
+
+template <typename T>
+void TestReduceOpQuantized(const int* input_dims_data, const float* input_data,
+                           float input_scale, int input_zero_point,
+                           const int* axis_dims_data, const int32_t* axis_data,
+                           const int* output_dims_data,
+                           const float* expected_output_data,
+                           float output_scale, int output_zero_point,
+                           TfLiteRegistration registration,
+                           TfLiteReducerParams* params) {
+  // Convert dimesion arguments to TfLiteArrays
+  TfLiteIntArray* input_dims = IntArrayFromInts(input_dims_data);
+  TfLiteIntArray* axis_dims = IntArrayFromInts(axis_dims_data);
+  TfLiteIntArray* output_dims = IntArrayFromInts(output_dims_data);
+
+  // Get number of elements in input and output tensors
+  const int output_dims_count = ElementCount(*output_dims);
+  const int input_dims_count = ElementCount(*input_dims);
+
+  // Allocate arrays for quantized tensors
+  T output_data_quant[output_dims_count];
+  T input_data_quant[input_dims_count];
+  T expected_output_data_quant[output_dims_count];
```

Comment:
same comment. since these array sizes are not known at compile time, this code is technically non-compliant. Please add these to the parameter list.

This next part of the comment is only for some context, no action needed from you:
The really long list of params that I am asking you to add to is not ideal at all -- we should be refactored into a struct or something else. But since this is a recurring theme in the TFLM testing code, its ok for you to make it even longer :-/